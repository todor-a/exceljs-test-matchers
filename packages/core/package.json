{
  "author": "Todor Andonov",
  "dependencies": {
    "table": "6.8.2"
  },
  "description": "",
  "devDependencies": {
    "vitest": "1.5.2"
  },
  "exports": {
    ".": {
      "import": {
        "default": "./build/index.js",
        "types": "./build/index.d.ts"
      },
      "require": {
        "default": "./build/index.cjs",
        "types": "./build/index.d.cts"
      }
    },
    "./package.json": "./package.json"
  },
  "files": [
    "build",
    "src"
  ],
  "keywords": [],
  "license": "ISC",
  "main": "build/index.cjs",
  "module": "build/index.js",
  "name": "@exceljs-test-matchers/core",
  "scripts": {
    "build": "pnpm run build:tsup",
    "build:tsup": "tsup",
    "clean": "rimraf ./build && rimraf ./coverage",
    "clean:all": "pnpm run clean && rimraf ./node_modules",
    "test:cspell": "cspell '**'",
    "test:eslint": "eslint --ext .ts,.tsx ./src",
    "test:types": "tsc",
    "test:unit": "vitest",
    "test:attw": "attw --pack"
  },
  "sideEffects": false,
  "type": "module",
  "types": "build/index.d.ts",
  "version": "1.0.0"
}
