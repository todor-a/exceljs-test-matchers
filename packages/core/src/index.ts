import * as ExcelJs from 'exceljs';

export const getSheetNames = (workbook: ExcelJs.Workbook) => {
    return workbook.worksheets.map(sheet => sheet.name);
};

export * from './findSheet';
export * from './hasSheetHeaders';
export * from './hasSheets';
export * from './isWorkbook';
export * from './toTable';
export * from './toWorkbook';
export * from './types';
