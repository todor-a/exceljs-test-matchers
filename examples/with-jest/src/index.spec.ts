import { toWorkbook } from '@exceljs-test-matchers/core';

describe('toMatchSheetSnapshot', () => {
    it('should be able to use the matcher', () => {
        const wb = toWorkbook([
            ['a', 'b', 'c'],
            ['1', '2', '3'],
            ['2', '3', '1'],
        ], 'sheet-a');

        expect(wb.getWorksheet('sheet-a')).toMatchSheetSnapshot();
    });
});

describe('toHaveSheets', () => {
    it('should be able to use the matcher', () => {
        const wb = toWorkbook([
            ['a', 'b', 'c'],
            ['1', '2', '3'],
            ['2', '3', '1'],
        ], 'sheet-a');

        expect(wb).toHaveSheets(['sheet-a']);
    });

    it('should fail when sheet does not exist', () => {
        const wb = toWorkbook([
            ['a', 'b', 'c'],
            ['1', '2', '3'],
            ['2', '3', '1'],
        ], 'sheet-a');

        try {
            expect(wb).toHaveSheets(['sheet-b']);
        } catch (error) {
            expect(error).toMatchInlineSnapshot(
                `[Error: Expected the sheets ["sheet-b"] to be present, but ["sheet-b"] are missing.]`,
            );
        }
    });

    it('should fail when sheet exist but is negated', () => {
        const wb = toWorkbook([
            ['a', 'b', 'c'],
            ['1', '2', '3'],
            ['2', '3', '1'],
        ], 'sheet-a');

        try {
            expect(wb).not.toHaveSheets(['sheet-a']);
        } catch (error) {
            expect(error).toMatchInlineSnapshot(
                `[Error: Expected the sheets ["sheet-a"] not to be present, but they were found.]`,
            );
        }
    });
});

describe('toHaveSheetHeaders', () => {
    const wb = toWorkbook([
        ['a', 'b', 'c'],
    ], 'sheet-a');

    it('should be able to use the matcher', () => {
        const ws = wb.getWorksheet('sheet-a');

        expect(ws).toHaveSheetHeaders(['a']);
    });

    it('should fail when sheet does not exist', () => {
        const ws = wb.getWorksheet('sheet-a');
        try {
            expect(ws).toHaveSheetHeaders(['w']);
        } catch (error) {
            expect(error).toMatchInlineSnapshot(`[Error: Expected the sheet headers ["w"] to be present, but ["w"] are missing.]`);
        }
    });

    it('should fail when sheet exist but is negated', () => {
        const ws = wb.getWorksheet('sheet-a');
        try {
            expect(ws).not.toHaveSheetHeaders(['a']);
        } catch (error) {
            expect(error).toMatchInlineSnapshot(`[Error: Expected the sheet headers ["a"] not to be present, but they were found.]`);
        }
    });
});
