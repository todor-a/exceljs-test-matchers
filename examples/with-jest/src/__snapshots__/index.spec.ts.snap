// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toMatchSheetSnapshot should be able to use the matcher 1`] = `
"
┌───┬───┬───┐
│ a │ b │ c │
├───┼───┼───┤
│ 1 │ 2 │ 3 │
├───┼───┼───┤
│ 2 │ 3 │ 1 │
└───┴───┴───┘
"
`;
